src/app/hooks/useAcceptTerms.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect } from "react";
+import { useLocalStorage } from "usehooks-ts";
+
+import { postTerms } from "@/app/api/postTerms";
+import { useError } from "@/app/context/Error/ErrorContext";
+import { ErrorState } from "@/app/types/errors";
+
+export const useAcceptTerms = (address: string, publicKeyNoCoord: string) => {
+  const [termsAcceptedAddresses, setTermsAcceptedAddresses] = useLocalStorage<
```

Comment:
If we want to use the local storage, we could associate the local storage entry with the address. However, I think that the safest approach is to just send the request every time the user accepts the terms (as we show them each time the user refreshes the page). Do we think that this is going to significantly increase our storage concerns or load? I think it would not have a major impact.